### DISCUSSION TITLE
Error with Firefox when starting Ghosts Client v7 - and other separate issues :/

### DISCUSSION THREAD

**Discussion by tm6907** [cmu-sei/GHOSTS#314] [Q&A] [answered]
Starting the Ghosts Client on an Ubuntu box using dotnet on the .dll file returns "Firefox profile cannot be loaded. it may be missing or inaccessible". Note: Im still using the v7 of Ghosts, I haven't updated all my other clients and API yet.

Another error I get when running "firefox" or the Ghosts client is "Not loading module "atk-bridge": The functionality is provided by GTK natively. Please try to not load it". However that doesn't seem to prevent firefox from opening still. 

I've tried a bunch of stuff online to fix it but nothings worked so I'm wondering if its due to Ghosts or the Geckodriver in someway.

![image](https://github.com/cmu-sei/GHOSTS/assets/162140616/91a10176-059a-46e3-b705-ac3bbc778f35)

The error just repeats, it never crashes. 

I'm having some other issues with a Kali-linux and Win7 client as well...
With my Win7 client, you start the ghosts.exe and it just holts in the cmd never progressing or starting activity with errors in the logs either - it was working before as well. 
With my Kali-linux client its running Ghosts and browsing sites, but isn't checking in despite being linked to the API the same way my other working clients are. 

Thanks

**Comment by sei-dupdyke** [comment]
Hmm, I have not seen that particular error. I would start by making sure the version of geckodriver that you are using aligns with the version of firefox. I think they manage releases on github now here: https://github.com/mozilla/geckodriver/releases

I believe I update to the current everytime I make an update, so it is possible you have multiple point versions but that they have very different versions of geckodriver supplied with them.

Sorry for the confusion.

**Reply by tm6907** [reply]
Yeah the version of geckodriver packaged with the Ghosts Client is an .exe so I downloaded geckodriver from that same link. I'm sure I have the correct version because its the same CPU architecture as the Rocky Linux box I have working with the client installed already. 

I have noticed the same error displays when running Ghosts on my working Rocky Linux box but doesn't prevent it running and starting activity anyway. 

Any advice for the issues with my other installs?

**Reply by sei-dupdyke** [reply]
If you take Ghosts out of the equation, and you start and run Firefox directly — does it run without issue? Can I see your timeline file that you are currently using? I'll try and run it here to recreate the issue.

> With my Win7 client, you start the ghosts.exe and it just holts in the cmd never progressing or starting activity with errors in the logs either - it was working before as well.

That sounds like an issue with the timeline. If it can't load the default timeline (misconfigured, etc. — which is easy to do, one wrong "}" gets me quite often). Try running the exe with:

cd wherever/ghosts/is/installed
ghosts.exe --debug

> With my Kali-linux client its running Ghosts and browsing sites, but isn't checking in despite being linked to the API the same way my other working clients are.

Anything in logs/app.log?

**Reply by tm6907** [reply]
Ok got the Win7 client to check in just waiting to see if it will browse Firefox now.  Edit: ran the .exe in debug like you recommend and found a issue. :) sorry not great at all this. "Network path not found" it is networked via a proxy and will browse internet fine. 
![image](https://github.com/cmu-sei/GHOSTS/assets/162140616/3d5e9404-7cf7-4014-b16c-037cabfa8d96)

Re: The Kali-linux box I've put in a snippet of the end of the log file. Sorry I can't upload the full file as is in a separate environment in a VM. The timeline is as-is from the download.
![image](https://github.com/cmu-sei/GHOSTS/assets/162140616/73bd3ba9-b689-4e8d-9fb0-90996d6e909f)

Re: The Ubuntu box, the timeline is all default as-is from the download. Running firefox will open and run fine but I do get this error however doesn't prevent it working. 
![image](https://github.com/cmu-sei/GHOSTS/assets/162140616/34aed88c-f2dc-47e7-baa6-66c912390056)

**Reply by sei-dupdyke** [reply]
I don't see anything of issue with the Win7 client. Are you saying it is not browsing the internet? What does the log/app.log file say?

On the kali-linux box, it kind of looks like that box doesn't have connectivity? "about:neterror" is usually an indication of this.

The ubuntu box looks to have an issue with firefox directly, none of those errors look to be related to ghosts.

So, I think you should establish that you can go on to each of these machines and browse in firefox to all of the sites you'd expect ghosts to browse to, and confirm it is not something that underlies the ghosts client, then we can come back and start to look at each of these clients on these different platforms. Sorry.

**Reply by tm6907** [reply]
Hi sorry for the late reply. I've given up with the Kali and Ubuntu client install for now. I'm gonna test the win7 box, last I tried it was running and connected to the internet but wasn't sure if I was reporting activity to the API. 

Separate question, is there a way to reset the ID numbering on the groups table within the PostGres database? 
Through my testing I've been removing/restarting clients and groups to test things and the number carries on incrementing. Searching for group ID '1' yields nothing since I removed it a while back, so any new client/group should default to '1' but doesn't. 

<img width="384" alt="image" src="https://github.com/cmu-sei/GHOSTS/assets/162140616/7792565e-9e66-45d9-917f-506eb1cdec44">

**Reply by sei-dupdyke** [reply, marked as answer]
That is an auto-generated Id, so you cannot edit it, sorry. Perhaps I should make this an unchanging guid in the future so that referencing it is far easier.
